# ***************************************************************************
# Copyright (c) 2017-2020 Siemens AG  All rights reserved.
# ***************************************************************************/
#
#
# Redistribution and use in source and binary forms, with or without
# modification, are permitted provided that the following conditions are met:
#
# * Redistributions of source code must retain the above copyright notice,
#   this list of conditions and the following disclaimer.
#
# * Redistributions in binary form must reproduce the above copyright notice,
#   this list of conditions and the following disclaimer in the documentation
#   and/or other materials provided with the distribution.
#
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
# ARE DISCLAIMED. IN NO EVENT SHALL Siemens AG OR CONTRIBUTORS BE LIABLE
# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
# DAMAGE.
# 
# Author(s): Thomas Riedmaier
--- ext/drcovlib/drcovlib.h
+++ ext/drcovlib/drcovlib.h
@@ -135,6 +135,7 @@
     uint start; /* offset of bb start from the image base */
     ushort size;
     ushort mod_id;
+    uint hits_since_last_reset; //CSA modification
 } bb_entry_t;
 
 /***************************************************************************
@@ -388,6 +389,10 @@
 
 /*@}*/ /* end doxygen group */
 
+//Defined by CSA
+DR_EXPORT drcovlib_status_t reset_coverage(void);
+DR_EXPORT drcovlib_status_t dump_current_coverage(void);
+
 #ifdef __cplusplus
 }
 #endif
